### PR TITLE
Docker Compose 3.8

### LINKED ISSUE #12428: Update all the Docker Compose files to Compose v3
All our Docker Compose files use the Compose v2 specification.

Current Compose version is v3.8 and we should update all of them -> https://docs.docker.com/compose/compose-file/

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #12428 

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: test/templates/compose/01-gateway/src/main/docker/app.yml [hunk 1]
```diff
@@ -9,10 +9,34 @@ services:
         ports:
             - 8080:8080
     jhgate-mysql:
-        extends:
-            file: mysql.yml
-            service: jhgate-mysql
+        image: mysql:5.7.18
```
Comment (DanielFran) [inline test/templates/compose/01-gateway/src/main/docker/app.yml:12]: Use constant: <%= DOCKER_MYSQL %>

File: test/templates/compose/02-mysql/src/main/docker/app.yml [hunk 2]
```diff
@@ -7,10 +7,34 @@ services:
             - SPRING_CLOUD_CONFIG_URI=http://admin:admin@jhipster-registry:8761/config
             - SPRING_DATASOURCE_URL=jdbc:mysql://msmysql-mysql:3306/msmysql?useUnicode=true&characterEncoding=utf8&useSSL=false&createDatabaseIfNotExist=true
     msmysql-mysql:
-        extends:
-            file: mysql.yml
-            service: msmysql-mysql
+        image: mysql:5.7.18
```
Comment (DanielFran) [inline test/templates/compose/02-mysql/src/main/docker/app.yml:10]: Use constant: <%= DOCKER_MYSQL %>

File: test/templates/compose/03-psql/src/main/docker/app.yml [hunk 3]
```diff
@@ -8,14 +8,39 @@ services:
             - SPRING_DATASOURCE_URL=jdbc:postgresql://mspsql-postgresql:5432/mspsql
             - SPRING_ELASTICSEARCH_REST_URIS=http://mspsql-elasticsearch:9200
     mspsql-postgresql:
-        extends:
-            file: postgresql.yml
-            service: mspsql-postgresql
+        image: postgres:9.5.1
```
Comment (DanielFran) [inline test/templates/compose/03-psql/src/main/docker/app.yml:11]: Use constant: <%= DOCKER_POSTGRESQL %>

File: test/templates/compose/03-psql/src/main/docker/app.yml [hunk 4]
```diff
@@ -8,14 +8,39 @@ services:
...
+        environment:
+            - POSTGRES_USER=mspsql
+            - POSTGRES_PASSWORD=
+        ports:
+            - 5432:5432
     mspsql-elasticsearch:
-        extends:
-            file: elasticsearch.yml
-            service: mspsql-elasticsearch
+        image: elasticsearch:2.4.6
```
Comment (DanielFran) [inline test/templates/compose/03-psql/src/main/docker/app.yml:20]: Use constant: <%= DOCKER_ELASTICSEARCH %>

File: test/templates/compose/04-mongo/src/main/docker/app.yml [hunk 5]
```diff
@@ -8,10 +8,29 @@ services:
             - SPRING_DATA_MONGODB_URI=mongodb://msmongodb-mongodb:27017
             - SPRING_DATA_MONGODB_DATABASE=msmongodb
     msmongodb-mongodb:
-        extends:
-            file: mongodb.yml
-            service: msmongodb-mongodb
+        image: mongo:4.2.3
```
Comment (DanielFran) [inline test/templates/compose/04-mongo/src/main/docker/app.yml:11]: Use constant: <%= DOCKER_MONGODB %>

File: test/templates/compose/05-cassandra/src/main/docker/app.yml [hunk 6]
```diff
@@ -8,20 +8,52 @@ services:
...
+        # volumes:
+        #     - ~/volumes/jhipster/mscassandra/cassandra/:/var/lib/cassandra/data
+        ports:
+            - 7000:7000
+            - 7001:7001
+            - 7199:7199
+            - 9042:9042
+            - 9160:9160
     mscassandra-cassandra-node:
         image: cassandra:2.2.5
```
Comment (DanielFran) [inline test/templates/compose/05-cassandra/src/main/docker/app.yml:25]: Use constant: <%= DOCKER_CASSANDRA %>

File: test/templates/compose/06-uaa/src/main/docker/app.yml [hunk 7]
```diff
@@ -7,10 +7,34 @@ services:
             - SPRING_CLOUD_CONFIG_URI=http://admin:admin@jhipster-registry:8761/config
             - SPRING_DATASOURCE_URL=jdbc:mysql://uaaserver-mysql:3306/uaaserver?useUnicode=true&characterEncoding=utf8&useSSL=false&createDatabaseIfNotExist=true
     uaaserver-mysql:
-        extends:
-            file: mysql.yml
-            service: uaaserver-mysql
+        image: mysql:5.7.18
```
Comment (DanielFran) [inline test/templates/compose/06-uaa/src/main/docker/app.yml:10]: Use constant: <%= DOCKER_MYSQL %>

File: test/templates/compose/07-mariadb/src/main/docker/app.yml [hunk 8]
```diff
@@ -7,10 +7,34 @@ services:
             - SPRING_CLOUD_CONFIG_URI=http://admin:admin@jhipster-registry:8761/config
             - SPRING_DATASOURCE_URL=jdbc:mariadb://msmariadb-mariadb:3306/msmariadb
     msmariadb-mariadb:
-        extends:
-            file: mariadb.yml
-            service: msmariadb-mariadb
+        image: mariadb:10.3.7
```
Comment (DanielFran) [inline test/templates/compose/07-mariadb/src/main/docker/app.yml:10]: Use constant: <%= DOCKER_MARIADB %>

File: test/templates/compose/08-monolith/src/main/docker/app.yml [hunk 9]
```diff
@@ -9,6 +9,14 @@ services:
         ports:
             - 8080:8080
     samplemysql-mysql:
-        extends:
-            file: mysql.yml
-            service: samplemysql-mysql
+        image: mysql:5.7.18
```
Comment (DanielFran) [inline test/templates/compose/08-monolith/src/main/docker/app.yml:12]: Use constant: <%= DOCKER_MYSQL %>

File: test/templates/compose/09-kafka/src/main/docker/app.yml [hunk 10]
```diff
@@ -17,16 +17,29 @@ services:
         ports:
             - 8080:8080
     samplekafka-mysql:
-        extends:
-            file: mysql.yml
-            service: samplekafka-mysql
+        image: mysql:5.7.20
```
Comment (DanielFran) [inline test/templates/compose/09-kafka/src/main/docker/app.yml:20]: Use constant: <%= DOCKER_MYSQL %>

File: test/templates/compose/09-kafka/src/main/docker/app.yml [hunk 11]
```diff
@@ -17,16 +17,29 @@ services:
...
+            - MYSQL_ALLOW_EMPTY_PASSWORD=yes
+            - MYSQL_DATABASE=samplekafka
+        ports:
+            - 3306:3306
+        command: mysqld --lower_case_table_names=1 --skip-ssl --character_set_server=utf8mb4 --explicit_defaults_for_timestamp
     kafka:
-        extends:
-            file: kafka.yml
-            service: kafka
+        image: confluentinc/cp-kafka:5.2.1
```
Comment (DanielFran) [inline test/templates/compose/09-kafka/src/main/docker/app.yml:31]: Use constant: <%= DOCKER_KAFKA %>

File: test/templates/compose/09-kafka/src/main/docker/app.yml [hunk 12]
```diff
@@ -17,16 +17,29 @@ services:
...
+            KAFKA_BROKER_ID: 2
+            KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+            KAFKA_ADVERTISED_HOST_NAME: kafka
+        ports:
+            - 9092:9092
     zookeeper:
-        extends:
-            file: kafka.yml
-            service: zookeeper
+        image: confluentinc/cp-zookeeper:5.2.1
```
Comment (DanielFran) [inline test/templates/compose/09-kafka/src/main/docker/app.yml:41]: Use constant: <%= DOCKER_ZOOKEEPER %>

File: test/templates/compose/11-mssql/src/main/docker/app.yml [hunk 13]
```diff
@@ -11,13 +11,36 @@ services:
             - SPRING_DATASOURCE_URL=jdbc:sqlserver://msmssql-mssql:1433;database=msmssql
             - JHIPSTER_SLEEP=30 # gives time for other services to boot before the application
     msmssql-mssql:
-        extends:
-            file: mssql.yml
-            service: msmssql-mssql
+        image: mcr.microsoft.com/mssql/server:2017-latest-ubuntu
```
Comment (DanielFran) [inline test/templates/compose/11-mssql/src/main/docker/app.yml:14]: Use constant: <%= DOCKER_MSSQL %>

**Review by DanielFran — CHANGES_REQUESTED** [review]
Since you are not using extends, should be deleted files:
- consul.yml.ejs
- jhipster-registry.yml.ejs
- kafka.yml.ejs
- keycloak.yml.ejs
- prometheus.yml.ejs
- traefik.yml.ejs ?

**Comment by pascalgrimaud** [comment]
I'll review and test it manually, self assign

**Comment by qmonmert** [comment]
@pascalgrimaud not finished yet

**Comment by qmonmert** [comment]
> Since you are not using extends, should be deleted files:
> 
> * consul.yml.ejs
> * jhipster-registry.yml.ejs
> * kafka.yml.ejs
> * keycloak.yml.ejs
> * prometheus.yml.ejs
> * traefik.yml.ejs ?

@DanielFran done

**Review by pascalgrimaud — APPROVED** [review]

**Comment by pascalgrimaud** [comment]
Just tested it, and everything works fine:

```
17:48:00 in tmp/20-microservices/compose took 2m 52s 
➜ dpss
NAMES                            IMAGE                               STATUS              PORTS
compose_invoice_1                invoice                             Up 2 minutes        5701/udp, 8082/tcp
compose_product_1                product                             Up 2 minutes        5701/udp, 8081/tcp
compose_jhipster-registry_1      jhipster/jhipster-registry:v6.3.0   Up 2 minutes        0.0.0.0:8761->8761/tcp
compose_store-postgresql_1       postgres:12.4                       Up 2 minutes        5432/tcp
compose_invoice-postgresql_1     postgres:12.4                       Up 2 minutes        5432/tcp
compose_store_1                  store                               Up 2 minutes        5701/udp, 0.0.0.0:8080->8080/tcp
compose_notification_1           notification                        Up 2 minutes        8083/tcp
compose_notification-mongodb_1   mongo:4.4.0                         Up 2 minutes        27017/tcp
compose_product-postgresql_1     postgres:12.4                       Up 2 minutes        5432/tcp
```

Then, there will be more tests with the JDL Daily Builds, tomorrow :)

**Pull request merged**